src/app.ts

```diff
@@ -13,14 +13,85 @@ interface TemperatureSummary {
   average: number
 }
 
+interface daySumary {
+  time: string
+  temperature: number
+  city: string
+  arrivalNumber: number
+}
+
+const listDaySumary: daySumary[][] = [];
+let sumaryOfDay: daySumary[] = [];
+let tempReading: TemperatureReading;
+let order: number;
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  tempReading = readings[0];
+  order = 1;
+  readings.forEach((dailyTemperature) => {
+    if (
+      dailyTemperature.time.toLocaleDateString() ===
+        tempReading.time.toLocaleDateString() &&
+      dailyTemperature.city === tempReading.city
+    ) {
+      sumaryOfDay.push({
+        time: dailyTemperature.time.toLocaleDateString(),
+        temperature: dailyTemperature.temperature,
+        city: dailyTemperature.city,
+        arrivalNumber: order,
+      })
+    } else {
+      listDaySumary.push(sumaryOfDay);
+      order = 1;
+      tempReading = dailyTemperature;
+      sumaryOfDay = [];
+      sumaryOfDay.push({
+        time: dailyTemperature.time.toLocaleDateString(),
+        temperature: dailyTemperature.temperature,
+        city: dailyTemperature.city,
+        arrivalNumber: order,
+      });
+    }
+    order++;
+  })
+  listDaySumary.push(sumaryOfDay);
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
+  const tempDate: string = date.toLocaleDateString();
+  const tempCity: string = city;
+  let sumTemp = 0;
+  const tempSumary: TemperatureSummary = {
+    first: 0,
+    last: 0,
+    high: -Infinity,
+    low: Infinity,
+    average: 0,
+  }
+
+  for (let i = 0; i < listDaySumary.length; i++) {
+    if (
+      listDaySumary[i][0].time == tempDate &&
+      listDaySumary[i][0].city == tempCity
+    ) {
+      for (let j = 0; j < listDaySumary[i].length; j++) {
+        if (listDaySumary[i][j].temperature > tempSumary.high) {
```

Comment:
Maybe you could use `Math.max()` instead of `if`

Reply:
`tempSummary.high = Math.max(listDaySumary[i][j].temperature, tempSumary.high)`

Reply:
thanks guys, i didn't know i could use the Math functions to not use an if structure